packages/eas-cli/src/build/android/applicationId.ts

```diff
@@ -1,191 +1,114 @@
-        }
-      }
+    const errorMessage = 'Could not read application id from Android project.';
+    const buildGradlePath = AndroidConfig.Paths.getAppBuildGradle(projectDir);
+    if (!fs.pathExistsSync(buildGradlePath)) {
+      throw new Error(errorMessage);
     }
-  } else if (!applicationIdFromAndroidProject && !applicationIdFromConfig) {
-    throw new Error(missingPackageMessage(configDescription));
-  } else if (applicationIdFromAndroidProject && !applicationIdFromConfig) {
-    if (getConfigFilePaths(projectDir).staticConfigPath) {
-      await updateAppJsonConfigAsync(projectDir, exp, applicationIdFromAndroidProject);
+    const buildGradle = fs.readFileSync(buildGradlePath, 'utf8');
+    const matchResult = buildGradle.match(/applicationId ['"](.*)['"]/);
+    return nullthrows(matchResult?.[1], errorMessage);
+  } else {
+    const applicationId = AndroidConfig.Package.getPackage(exp);
+    if (!applicationId || !isApplicationIdValid(applicationId)) {
+      if (applicationId) {
+        Log.warn(INVALID_APPLICATION_ID_MESSAGE);
+      }
+      throw new Error(
+        `Specify "android.package" in ${getProjectConfigDescription(
+          projectDir
+        )} and run this command again.`
+      );
     } else {
-      throw new Error(missingPackageMessage(configDescription));
+      return applicationId;
     }
-  } else if (!applicationIdFromAndroidProject && applicationIdFromConfig) {
-    // This should never happen, adding warning just in case
-    Log.warn(
-      'applicationId is not specified in your ./android/app/build.gradle file. Make sure your project is configured correctly before building.'
-    );
-    return;
-  }
-  if (allowExperimental) {
-    // this step is optional, the Play store is using applicationId value from build.gradle
-    // to identify the app, so package name does not need to be updated
-    await updatePackageNameAsync(projectDir, exp);
-    await gitAddAsync(path.join(projectDir, 'android'), { intentToAdd: true });
   }
 }
 
-function missingPackageMessage(configDescription: string): string {
-  return `Please define "android.package" in ${configDescription} and run "eas build:configure" again.`;
-}
-
-async function updateAppJsonConfigAsync(
-  projectDir: string,
-  exp: ExpoConfig,
-  newApplicationId: string
-): Promise<void> {
+async function configureApplicationIdAsync(projectDir: string, exp: ExpoConfig): Promise<string> {
   const paths = getConfigFilePaths(projectDir);
-  assert(paths.staticConfigPath, "can't update dynamic configs");
+  // we can't automatically update app.config.js
+  if (!paths.staticConfigPath) {
+    throw new Error(
+      `"android.package" is not defined in your app.config.js and we can't update this file programatically. Add the value on your own and run this command again.`
+    );
+  }
+
+  const { packageName } = await promptAsync({
+    name: 'packageName',
+    type: 'text',
+    message: `What would you like your Android package name to be?`,
```

Comment:
android package is not really correct, we use field `package` in app.json but it's applicationId we want to define here